spark/src/test/scala/org/apache/spark/sql/delta/DeltaParquetFileFormatSuite.scala

```diff
@@ -16,140 +16,214 @@
+            if (enableDVs) {
+              // Select some rows that are deleted and some rows not deleted
+              // Deleted row `value`: 0, 200, 300, 756, 10352, 19999
+              // Not deleted row `value`: 7, 900
+              checkDatasetUnorderly(
+                Dataset.ofRows(spark, plan)
+                  .filter("value in (0, 7, 200, 300, 756, 900, 10352, 19999)")
+                  .select("value", DeltaParquetFileFormat.IS_ROW_DELETED_COLUMN_NAME)
+                  .as[(Int, Int)],
+                (0, deletedColumnValue),
+                (7, notDeletedColumnValue),
+                (200, deletedColumnValue),
+                (300, deletedColumnValue),
+                (756, deletedColumnValue),
+                (900, notDeletedColumnValue),
+                (10352, deletedColumnValue),
+                (19999, deletedColumnValue))
+            } else {
+              checkDatasetUnorderly(
+                Dataset.ofRows(spark, plan)
+                  .filter("value in (0, 7, 200, 300, 756, 900, 10352, 19999)")
+                  .select("value", DeltaParquetFileFormat.IS_ROW_DELETED_COLUMN_NAME)
+                  .as[(Int, Int)],
+                (0, notDeletedColumnValue),
+                (7, notDeletedColumnValue),
+                (200, notDeletedColumnValue),
+                (300, notDeletedColumnValue),
+                (756, notDeletedColumnValue),
+                (900, notDeletedColumnValue),
+                (10352, notDeletedColumnValue),
+                (19999, notDeletedColumnValue))
+            }
           }
 
-          val df = Dataset.ofRows(spark, plan)
-          assert(rowIndexes(df) === Seq.range(0, 20000).toSet)
+          if (readRowIndexCol) {
+            def rowIndexes(df: DataFrame): Set[Long] = {
+              val colIndex = if (readIsRowDeletedCol) 2 else 1
+              df.collect().map(_.getLong(colIndex)).toSet
+            }
+
+            val df = Dataset.ofRows(spark, plan)
+            assert(rowIndexes(df) === Seq.range(0, 20000).toSet)
 
-          assert(
-            rowIndexes(
-              df.filter("value in (0, 7, 200, 300, 756, 900, 10352, 19999)")) ===
-            Seq(0, 7, 200, 300, 756, 900, 10352, 19999).toSet)
+            assert(
+              rowIndexes(
+                df.filter("value in (0, 7, 200, 300, 756, 900, 10352, 19999)")) ===
+                Seq(0, 7, 200, 300, 756, 900, 10352, 19999).toSet)
+          }
         }
       }
     }
   }
 
+  testWithBothParquetReaders("Existence of Deletion Vectors metadata columns when we " +
+    "add and remove DV from the table's properties") {
+    withTempDir { tempDir =>
+      val tablePath = tempDir.toString
+      val deltaLog = DeltaLog.forTable(spark, tempDir)
+
+      // Version 1: Create a table without DV and insert some values.
+      generateData(tablePath)
+
+      checkDVMetadataFields(tablePath, checkMetadataFields = false)
```

Comment:
Can we also check the table contents using `checkResult` each time after we check the metadata?